gateway/reverse_proxy.go

```diff
@@ -1220,13 +1220,17 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 	}
 
 	p.addAuthInfo(outreq, req)
+	err := p.addUpstreamOAuthInfo(outreq, req)
+	if err != nil {
+		p.ErrorHandler.HandleError(rw, logreq, err.Error(), 500, true)
```

Comment:
This should be performed earlier, a-la middleware, https://github.com/TykTechnologies/tyk/pull/6596

Reply:
Discussed on slack that the required functionality cannot be achieved via middlewares (also discussed with Leo)